dialer/dialer.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dialer contains methods and structs for creating secure, authorized connections to a Cloud SQL instance.
+package dialer
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/kurtisvg/cloud-sql-go-connector/internal/cloudsql"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+var (
+	once sync.Once
+	d    *dialManager
```

Comment:
Should this be `dm`  to avoid the shadowing  that happens on l.39 below and where a `d` receiver is declared?

Reply:
Good catch. 

There was actually a shadow that should have happened but didn't on L73 (that would have lead to a bug).